codersdk/name_test.go

```diff
@@ -254,3 +256,40 @@ func TestUserRealNameValid(t *testing.T) {
 		})
 	}
 }
+
+func TestGroupNameValid(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		Name  string
+		Valid bool
+	}{
+		{"", false},
+		{"my-group", true},
+		{"create", false},
+		{"new", false},
+		{"Lord Voldemort Team", false},
+		{randomString(255), true},
+		{randomString(256), false},
+	}
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.Name, func(t *testing.T) {
+			t.Parallel()
+			err := codersdk.GroupNameValid(testCase.Name)
+			assert.Equal(t, testCase.Valid, err == nil)
```

Comment:
Can you modify this to contain the error if something bad happens:

`assert.Equal(t, testCase.Valid, err == nil, "Test case %s failed: expected valid=%t but got error: %v", testCase.Name, testCase.Valid, err)`